waltz-tools/src/main/java/com/wepay/waltz/tools/server/ServerCli.java

```diff
@@ -279,16 +282,18 @@ protected void processCmd(CommandLine cmd) throws SubCommandFailedException {
                 String host = hostAndPortArray[0];
                 int serverPort = Integer.parseInt(hostAndPortArray[1]);
 
-                if (!partitionId.matches("^[0-9]+$")) {
-                    throw new IllegalArgumentException(String.format("Partition id '%s' is invalid. Expected a non-negative integer", partitionId));
-                }
-                removePreferredPartition(host, serverPort, Integer.parseInt(partitionId), cliConfigPath);
+                List<Integer> partitionIds = CliUtils.parseIntRanges(partitionId);
+                removePreferredPartition(host, serverPort, partitionIds, cliConfigPath);
             } catch (Exception e) {
                 throw new SubCommandFailedException(String.format("Failed to remove preferred partition .%n%s", e));
             }
         }
 
-        private void removePreferredPartition(String host, int serverPort, int partitionId, String cliConfigPath) {
+        private void removePreferredPartition(String host, int serverPort, List<Integer> partitionIds, String cliConfigPath) {
+            if (partitionIds.size() == 0) {
+                throw new IllegalArgumentException("Partitions array is empty");
+            }
+
```

Comment:
Same as the above.
Can we also validate the min and max partition id from the given list and verify it is as expected please. Because, we should fail immediately if the partitionId is invalid rather than passing an invalid partition to the server.